synapseutils/sync.py

```diff
@@ -111,14 +113,18 @@ def generateManifest(syn, allFiles, filename):
 
     :param filename: file where manifest will be written
     """
-    keys = ['path', 'parent', 'name', 'synapseStore', 'contentType', 'used',
-            'executed', 'activityName', 'activityDescription']
+    keys, data = _process_manifest_rows(syn, allFiles)
+    _write_manifest_data(filename, keys, data)
+
+
+def _process_manifest_rows(syn, allFiles):
+    keys = list(DEFAULT_MANIFEST_KEYS)
     annotKeys = set()
     data = []
     for entity in allFiles:
         row = {'parent': entity['parentId'], 'path': entity.get("path"), 'name': entity.name,
-               'synapseStore': entity.synapseStore, 'contentType': allFiles[0]['contentType']}
-        row.update({key:(val[0] if len(val) > 0 else "") for key, val in entity.annotations.items()})
+               'synapseStore': entity.synapseStore, 'contentType': entity['contentType']}
+        row.update({key: (val[0] if len(val) > 0 else "") for key, val in entity.annotations.items()})
 
         annotKeys.update(set(entity.annotations.keys()))
         try:
```

Comment:
This block of code could be refactor to a helper method whose contract is to return a valid value. This helper should handle the error appropriately. There should be unit test for this helper.